tests/integration/security/tls_origination/egress_gateway_origination_test.go

```diff
@@ -0,0 +1,482 @@
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package egressgatewayorigination
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"path"
+	"reflect"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework/resource"
+
+	"io/ioutil"
+
+	envoyAdmin "github.com/envoyproxy/go-control-plane/envoy/admin/v3"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/echo/common"
+	"istio.io/istio/pkg/test/echo/common/response"
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource/environment"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/structpath"
+)
+
+func mustReadCert(t *testing.T, f string) string {
+	b, err := ioutil.ReadFile(path.Join(env.IstioSrc, "tests/testdata/certs", f))
+	if err != nil {
+		t.Fatalf("failed to read %v: %v", f, err)
+	}
+	return string(b)
+}
+
+const (
+	strictMTlsConfig = "testdata/strict-mtls.yaml"
+)
+
+// TestEgressGatewayTls brings up an cluster and will ensure that the TLS origination at
+// egress gateway allows secure communication between the egress gateway and external workload.
+// This test brings up an egress gateway to originate TLS connection. The test will ensure that requests
+// are routed securely through the egress gateway and that the TLS origination happens at the gateway.
+func TestEgressGatewayTls(t *testing.T) {
+	framework.NewTest(t).
+		Features("security.egress.tls").
```

Comment:
change Features("security.egress.tls"). to Features("security.egress.tls.filebased"). The feature label is still "security.egress.tls", but you can append a scenario to it.
You need to add feature label into pkg/test/framework/features/features.yaml. You can refer to https://github.com/istio/istio/pull/24346.